### PR TITLE
test: unify using `@pytest.mark.flaky(...)` & bump pytest plugins

### PR DESCRIPTION
**Pull request by Borda**
*Issue #, if available:*

*Description of changes:*
seem that there is mix of few packages and maybe better to use just `pytest` extension

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

cc: @lostella

### CODE REVIEW

**Comment by Borda** [comment]
switching to `pytest-rerunfailures` for py3.8 support and messing with some other stuff

**Comment by Borda** [comment]
following suggestion from https://github.com/pytest-dev/pytest/issues/11052#issuecomment-1567323056

File: .github/workflows/mxnet_nightly.yml [hunk 1]
```diff
@@ -36,4 +36,4 @@ jobs:
           -r requirements/requirements-extras-autogluon.txt
     - name: Test with pytest
       run: |
-        pytest -m 'not (gpu or serial)' --cov src/gluonts --cov-report=term --cov-report xml test
+        pytest -m 'not (gpu or serial)' --cov=gluonts --cov-report=term --cov-report xml test
```
Comment (lostella) [inline .github/workflows/mxnet_nightly.yml:39]: Why this change?

File: requirements/requirements-test.txt [hunk 2]
```diff
@@ -1,11 +1,11 @@
-pandas>=1.1
-flaky~=3.6
-pytest-cov==2.6.*
-pytest-timeout~=1.3
-pytest-xdist~=1.27
-pytest>=6.0
+pandas >=1.1
+pytest >7.0
+pytest-cov >4.0, <5.0
```
Comment (lostella) [inline requirements/requirements-test.txt]: Maybe use `~=4.0`.

Also why adding the spaces?

File: requirements/requirements-test.txt [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
+pytest-timeout >2.0, <3.0
+pytest-xdist >3.0, <4.0
+pytest-rerunfailures >=13.0, <14.0
 ujson
 orjson
 requests
-holidays~=0.9
-matplotlib~=3.6
+holidays >=0.9
```
Comment (lostella) [inline requirements/requirements-test.txt]: Always better to have an upper bound to requirements versions. I know some other don’t have it, but maybe let’s not remove them from the ones that do.

File: requirements/requirements-test.txt [hunk 4]
```diff
@@ -1,11 +1,11 @@
...
+matplotlib >=3.6, <3.7
```
Comment (lostella) [inline requirements/requirements-test.txt]: `~=3.6`?

File: test/mx/distribution/test_mx_distribution_inference.py [hunk 5]
```diff
@@ -1291,7 +1291,7 @@ def test_inflated_poisson_likelihood(
 
 
 @pytest.mark.timeout(150)
-@pytest.mark.flaky(max_runs=6, min_passes=1)
+@pytest.mark.flaky(retries=3)
```
Comment (lostella) [inline test/mx/distribution/test_mx_distribution_inference.py]: Could you make sure that the number of trials is unchanged?

File: requirements/requirements-test.txt (hunk 3 above)
Comment (Borda) [reply requirements/requirements-test.txt]: ok, just the 0.10 was too old
```suggestion
holidays >=0.9, <=0.43
```

File: requirements/requirements-test.txt (hunk 4 above)
Comment (Borda) [reply requirements/requirements-test.txt]: I had some problem with it and `~` is conda notation not pip

File: test/mx/distribution/test_mx_distribution_inference.py (hunk 5 above)
Comment (Borda) [reply test/mx/distribution/test_mx_distribution_inference.py]: updated

File: .github/workflows/mxnet_nightly.yml (hunk 1 above)
Comment (Borda) [reply .github/workflows/mxnet_nightly.yml:39]: src is not the package name but the path and before you also install package with `pup install -e .` so src/ is not relevant

File: requirements/requirements-test.txt (hunk 2 above)
Comment (Borda) [reply requirements/requirements-test.txt]: I can prune pace, just finding it easier to read

Comment (Borda) [reply requirements/requirements-test.txt]:
```suggestion
pytest-cov >4.0,<5.0
```

File: test/mx/distribution/test_mx_distribution_inference.py [hunk 6]
```diff
@@ -1243,7 +1243,7 @@ def test_genpareto_likelihood(xi: float, beta: float, hybridize: bool) -> None:
 
 
 @pytest.mark.timeout(120)
-@pytest.mark.flaky(max_runs=6, min_passes=1)
+@pytest.mark.flaky(retries=9)
```
Comment (Borda) [inline test/mx/distribution/test_mx_distribution_inference.py]: Increasing to 9 as it is really flaky...
maybe consider some seeding?

**Comment by Borda** [comment]
@lostella, it seems like seeding tests also help to gain some stability...

File: requirements/requirements-test.txt (hunk 4 above)
Comment (lostella) [reply requirements/requirements-test.txt]: I think it’s more general than conda https://packaging.python.org/en/latest/specifications/version-specifiers/#id5

File: requirements/requirements-test.txt [hunk 7]
```diff
@@ -1,11 +1,11 @@
...
+pytest-cov >4.0,<5.0
+pytest-timeout >2.0, <3.0
+pytest-xdist >3.0, <4.0
+pytest-rerunfailures >=13.0, <14.0
 ujson
 orjson
 requests
-holidays~=0.9
-matplotlib~=3.6
+holidays >=0.9, <=0.43
```
Comment (lostella) [inline requirements/requirements-test.txt]:
```suggestion
holidays~=0.9
```

File: requirements/requirements-test.txt [hunk 8]
```diff
@@ -1,11 +1,11 @@
...
+matplotlib >=3.6, <3.7
```
Comment (lostella) [inline requirements/requirements-test.txt]:
```suggestion
matplotlib~=3.6
```

**Review by lostella — COMMENTED** [review]
@Borda I suggest to revert two changes (see inline proposed changes). In general, keeping PRs scoped to a small set of changes makes it easier to review, approve & merge faster

File: requirements/requirements-test.txt (hunk 4 above)
Comment (Borda) [reply requirements/requirements-test.txt]: So do you want me to use ~ for all?

**Comment by Borda** [comment]
@lostella seems all tests are :green_circle:

**Review by lostella — APPROVED** [review]
